### ISSUE TITLE
Java: Deprecate and remove Element::getDoc

### ISSUE THREAD

**Issue by Marcono1234** [github/codeql#5320]
`Element::getDoc` only performs an implicit cast of `this`:
https://github.com/github/codeql/blob/f9973d10aea9c550d91f6607b7cfdd2f543d64ee/java/ql/src/semmle/code/java/Element.qll#L42-L43

- This is probably redundant because you can instead write a cast `element.(Document) ...`
- It is irritating because it suggests that every `Element` can be 'converted' to `Document`, which is not the case
- It causes irritating error messages when used on types which cannot be cast to `Document`:  
![Error message screenshot](https://user-images.githubusercontent.com/11685886/109879540-fba75f00-7c75-11eb-86cd-b2775019dc81.png)

Therefore it might be good to deprecate and remove the predicate.

**Comment by aschackmull** [comment]
I agree that it would be nice to clean this up a bit, as `getDoc` indeed shouldn't be available on `Package`. However, we would like to have it available on both `Member`, `Class`, and `Interface`, and their nearest common ancestor is currently `Element` so that's why `getDoc` is defined there. Cleaning this up in a nice way requires a little bit thought and some work and is fairly low priority, so we may not immediately fix this.
